woocommerce/woocommerce
# Change Variations form shown in Variations tab when there are no variations created

### All Submissions:






### Changes proposed in this Pull Request:

Change Variations form shown in Variations tab when there are no variations created

I created a new ajax action called `add_attributes_and_variations` in order to reduce roundtrips and payload, instead of calling `add_attributes` and then adding variations.

Unfortunately, I had to manually reload the page and replace the variations and attributes box content with jQuery after the action is called. I'm not that happy with the solution but I don't see an alternative without a major refactoring of how things are structured.

I'm currently looking into how to remove duplicated code between `html-product-data-variations.php` and `html-product-attribute`.



Closes #36658.



### How to test the changes in this Pull Request:




1. Go to the current product add screen
2. Select "Variable Product"
3. Click on Variations tab
4. Enter name and some values
5. Click on "Save variations"
6. Check that variations were created for all the values
7. Check that attributes tab was correctly reloaded



### Other information:




### FOR PR REVIEWER ONLY:
